src/main/cpp/util/file_platform.h

```diff
@@ -31,10 +31,12 @@ class IFileMtime {
  public:
   virtual ~IFileMtime() {}
 
-  // Queries the mtime of `path` to see whether it's in the distant future.
-  // Returns true if querying succeeded and stores the result in `result`.
-  // Returns false if querying failed.
-  virtual bool GetIfInDistantFuture(const std::string &path, bool *result) = 0;
+  // Checks if `path` is a good "extracted binary", i.e. it is either an
+  // embedded binary that Bazel extracted from a previous run, and it hasn't
+  // been tampered with (its mtime is unchanged), or it's a directory.
+  // Returns false if querying the information failed, either due to I/O error
+  // or due to the `path` missing or it being a broken symlink.
+  virtual bool CheckExtractedBinary(const std::string &path, bool *result) = 0;
```

Comment:
I would personally put this in blaze_util_platform, if the comment is true and that its properties are quite Bazel-specific

Reply:
I agree, but moving it there would make this PR a lot more complicated. Let me not do that now.